ascent/src/rel_index_read.rs

```diff
@@ -15,6 +15,10 @@ pub trait RelIndexRead<'a>{
    type IteratorType: Iterator<Item = Self::Value> + Clone + 'a;
    fn index_get(&'a self, key: &Self::Key) -> Option<Self::IteratorType>;
    fn len(&self) -> usize;
+
+   fn is_empty(&self) -> bool {

```

Comment:
Is this function used anywhere?

Reply:
No, but clippy warns about "`len()` without `is_empty()` ([lint](https://rust-lang.github.io/rust-clippy/master/index.html#/len_without_is_empty)), since using `.len() == 0` has [another lint](https://rust-lang.github.io/rust-clippy/master/index.html#/len_zero) and the lack of `is_empty()` encourages that.

Reply:
I see. Can you silence the lint for now? This function should be renamed to something like `len_estimate()`, which would get rid of the warning too. (Please don't rename the function for now)

Reply:
Sure! Fixed.

Reply:
Thanks!